### ISSUE TITLE
[Feature Request] Filter by already downloaded (any criteria)

### ISSUE THREAD

**Issue by redeven** [erengy/taiga#943]
Hey! Recently, I'd been experiencing an issue where sometimes Taiga would download the same episode multiple times, either from different fansubs (if I didn't hard filter it to one choice) or different versions as they released (if I didn't filter that too), if those new torrents were uploaded before I watched the episode and the %watched% filter cleared them.

It may be an issue with my settings or poor filter planning, and if there's already a fix I'd love to fix it, but I also thought of the idea of being able to just say "if episode X of anime Y is already downloaded, regardless of other criteria like fansub, quality, version, etc, discard it".

I thought it could either be a standalone option or even just a new variable %downloaded% for the "Episode number" condition (returns either highest episode number found, and maybe 0 when none found?).

**Comment by paulo27ms** [comment]
This already exists, it's one of the default filters.

**Comment by redeven** [comment]
Oh I absolutely completely missed the Episode availability filter. Thank you!

**Comment by olee** [comment]
I think this is actually an issue, because recently I have the same issue that the same episode is downloaded 2-3 times in parallel, even though I didn't change anything in the configuration and the filter for episode availability is enabled.
I really need some help here as this is causing quite a lot of unnecessary network and disk load.
That's why I think we should reopen this issue.

**Comment by paulo27ms** [comment]
If you mean the exact same file that shouldn't happen at all (your client would recognize the file anyway), if you mean different versions your filter is wrong. Use the default filter.

**Comment by olee** [comment]
Not same file, but same episode.

**Comment by olee** [comment]
Also the default filter is enabled:
![image](https://user-images.githubusercontent.com/6547794/97058730-e49d8580-158e-11eb-94e3-23509201e404.png)

**Comment by paulo27ms** [comment]
Only other reason this could be happening is if you're saving to an external HDD and not doing a manual refresh when opening up Taiga. If that's not the case you'll need to replicate the issue somehow.

**Comment by olee** [comment]
No. As I said it was working **before**, and only **broke recently**.
What actually happens is that it starts downloading multiple torrents **at the same time**.
This should not happen afaik, as it should always only pick one version per episode, doesn't it?

**Comment by paulo27ms** [comment]
Then it's problem with the rest of your filters. All that default filter does is check episodes available and watched, if you haven't downloaded or watched it yet, of course it'll download as many as your filters allow it.

**Comment by olee** [comment]
I guess the problem happens when multiple anime are released at the same time. My torrent client is configured to have no more than 2 downloads active at the same time.
So maybe it tried to queue like 4 different series and while downloading it, new releases for one of those queued were discovered and also queued then.

It would be great if the downloader could be adjusted so that it does not try downloading the same episode again after it already started a download.
It could be some cache like a dictionary which sets a flag with a multi key like <anime, episode> and maybe as a value it could contain when it started the download so it does not retry for a while.
That would prevent such issues from happening.

**Comment by olee** [comment]
bump.

I can still confirm that this feature is not working as it used to.
I'm a coder myself and worked on many OSS projects so I know many people make pointless issues which were just them using the software incorrectly etc.
But in this case it's an actual regression which is why I told you it was not a problem with my filters (which I never touched at all when it was working before), but something else.

**Comment by paulo27ms** [comment]
I think it'd be best to create a new issue at this point as you're describing a new feature you want and not necessarily an error with the current system (I don't think filtered episodes by downloaded torrents was the goal for this filter).
